llvm/include/llvm/IR/IntrinsicsAArch64.td

```diff
@@ -2899,6 +2899,15 @@ let TargetPrefix = "aarch64" in {
           [llvm_i32_ty],
           [IntrNoMem, IntrHasSideEffects]>;
 
+  def int_aarch64_sme_write_lane_zt
```

Comment:
`svzero_zt` is defined like this:

`def int_aarch64_sme_zero_zt : DefaultAttrsIntrinsic<[], [llvm_i32_ty], [ImmArg<ArgIndex<0>>, IntrWriteMem]>;`

Apologies if I've already asked this, but should these intrinsics not also model writing to ZT0 as memory in the same way? (rather than InrtNoMem, IntrHasSideEffects)?

Reply:
So ZT0 is a register and not a memory.
 I believe the correct way to model this is with IntraHasSideEffects and not correct to model this as writing in memory.